### PR TITLE
Add base code

### PR DESCRIPTION
**Pull request by freya022**

### CODE REVIEW

**Comment by RGBCube** [comment]
I think we should just do a `config.json5`

**Comment by RGBCube** [comment]
I think we should just do a `config.json5`

**Review by Xirado — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/github/gasbarrel/commands/slash/SlashPing.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        event.jda.restPing
+            .await()
+            .let { event.reply_("Pong! $it ms", ephemeral = true) }
+            .queue()
+    }
+
+    @AppDeclaration
+    fun declare(manager: GlobalApplicationCommandManager) {
+        manager.slashCommand("ping", scope = CommandScope.GLOBAL) {
+            description = "Pong !"
```
Comment (TopiSenpai) [inline src/main/kotlin/io/github/gasbarrel/commands/slash/SlashPing.kt:24]:
```suggestion
            description = "Pong!"
```

File: src/main/kotlin/io/github/gasbarrel/Main.kt [hunk 2]
```diff
@@ -1,7 +1,76 @@
...
+
+private val logger by lazy { KotlinLogging.logger {} } // Must not load before system property is set
+
+fun main(args: Array<out String>) {
+    try {
+        Data.init()
+
+        if (Data.logbackConfigPath.exists()) {
+            System.setProperty(ClassicConstants.CONFIG_FILE_PROPERTY, Data.logbackConfigPath.absolutePathString())
+            logger.info( "Loading production logback config")
```
Comment (TopiSenpai) [inline src/main/kotlin/io/github/gasbarrel/Main.kt:26]:
```suggestion
            logger.info("Loading production logback config")
```

File: src/main/kotlin/io/github/gasbarrel/Main.kt [hunk 3]
```diff
@@ -1,7 +1,76 @@
...
+        } else {
+            logger.info( "Loading test logback config")
```
Comment (TopiSenpai) [inline src/main/kotlin/io/github/gasbarrel/Main.kt:28]:
```suggestion
            logger.info("Loading test logback config")
```

File: src/main/kotlin/io/github/gasbarrel/commands/slash/SlashPing.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.freya02.botcommands.api.annotations.CommandMarker
+import com.freya02.botcommands.api.commands.application.CommandScope
+import com.freya02.botcommands.api.commands.application.GlobalApplicationCommandManager
+import com.freya02.botcommands.api.commands.application.annotations.AppDeclaration
+import com.freya02.botcommands.api.commands.application.slash.GlobalSlashEvent
+import dev.minn.jda.ktx.coroutines.await
+import dev.minn.jda.ktx.messages.reply_
+
+@CommandMarker //Shut up IJ
```
Comment (TopiSenpai) [inline src/main/kotlin/io/github/gasbarrel/commands/slash/SlashPing.kt:11]:
```suggestion
@CommandMarker // Shut up IJ
```

File: src/main/kotlin/io/github/gasbarrel/JDAService.kt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import dev.minn.jda.ktx.jdabuilder.light
+import net.dv8tion.jda.api.entities.Activity
+import net.dv8tion.jda.api.hooks.IEventManager
+
+@BService(ServiceStart.READY)
+class JDAService(config: Config, manager: IEventManager) {
+    init {
+        light(config.token, enableCoroutines = false) {
+            setMaxReconnectDelay(128)
+            setActivity(Activity.playing("with a cigarette"))
```
Comment (TopiSenpai) [inline src/main/kotlin/io/github/gasbarrel/JDAService.kt:14]:
```suggestion
            setActivity(Activity.playing("with a cigarette 🚬"))
```

**Review by TopiSenpai — CHANGES_REQUESTED** [review]

**Review by Xirado — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/github/gasbarrel/Data.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import kotlin.io.path.Path
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+import kotlin.io.path.notExists
+
+// If test configs exists then they should be loaded
+// If not then fallback to the config path, which must be validated
+// Other configs such as logback are set in the bot folder
+object Data {
+    private val botFolder: Path = Path(System.getProperty("user.home"), "Bots", "GasBarrel").validate("Bot folder")
```
[the remark this answers is not in the document]
Comment (RGBCube) [reply src/main/kotlin/io/github/gasbarrel/Data.kt:14]: I think we should do `.config/GasBarrel` on Linux, `%APPDATA%\GasBarrel` on Windows and so on

[the remark this answers is not in the document]
Comment (RGBCube) [reply src/main/kotlin/io/github/gasbarrel/Data.kt]: We can do this by
```kt
val configFolder = System.getenv("XDG_CONFIG_HOME") ?: "${System.getProperty("user.home")}/.config"
```

**Review by Xirado — APPROVED** [review]

**Review by TopiSenpai — DISMISSED** [review]
kotlin bad, can't approve, rewrite in go

**Review by Xirado — APPROVED** [review]

**Pull request merged**